CairoMakie/src/primitives.jl

```diff
@@ -315,6 +317,24 @@ function draw_marker(ctx, marker::Rect, pos, scale, strokecolor, strokewidth, ma
 end
 
 
+function draw_marker(ctx, marker::Matrix{T}, pos, scale,
+        strokecolor #= unused =#, strokewidth #= unused =#,
+        marker_offset, rotation) where T<:Colorant
+
+    # convert marker to Cairo compatible image data
+    argb32_marker = convert.(ARGB32, marker)
+    argb32_marker = permutedims(argb32_marker, (2,1)) # swap x-y for Cairo
+    marker_surf   = Cairo.CairoImageSurface(argb32_marker)
+
+    Cairo.translate(ctx, pos[1]+marker_offset[1], pos[2]+marker_offset[2])
+    Cairo.rotate(ctx, to_2d_rotation(rotation))
+    px_scale = scale ./ size(argb32_marker)
+    Cairo.scale(ctx, px_scale[1], px_scale[2])
+    Cairo.set_source_surface(ctx, marker_surf, 0, 0)
```

Comment:
I think you might want to set something like:
```julia
    p = Cairo.get_source(ctx)
    Cairo.pattern_set_extend(p, Cairo.EXTEND_NONE)
    filt =  Cairo.FILTER_BILINEAR # or Cairo.FILTER_NEAREST, whichever seems best
    Cairo.pattern_set_filter(p, filt)
```

in order to get the sharpest image.

Reply:
Just tried this but it did not have any effect.
I also tried with adding
```julia
Cairo.fill(ctx)
Cairo.restore(ctx)
```
because this is also present in `draw_atomic` for `Heatmap, Image` where your code snippet is also used, but with no effect.

Also noticed that the doggy now has weirdly colored boundaries around its face: 
![image](https://user-images.githubusercontent.com/26469701/178805887-1e53c84b-308b-46ac-8923-4a1a9786e24a.png)

Is that supposed to be fixed by the filter? Or is this a different issue?

Reply:
Neverminde, the weird colors are due to me saving the figure as `.pdf`.
The edges disappear with `.png`.

Reply:
Huh...i guess those are there because of (a) weird rendering or (b) color interpolation.  Does the same thing happen if you open it in Google Chrome?

Reply:
It appears that the colored edges issue is not specific to the `scatter` implementation, since it also occurs when using an `image`.

I think we have to track this in a separate issue.

Reply:
Adding `fxaa=true` does not help either.